app/src/androidMain/kotlin/features/scan/ScanViewModel.kt

```diff
@@ -36,22 +33,22 @@ class ScanViewModel(application: Application) : AndroidViewModel(application) {
     private val scanScope = viewModelScope.childScope()
     private val found = hashMapOf<String, Advertisement>()
 
-    private val _scanStatus = MutableStateFlow<ScanStatus>(Stopped)
-    val scanStatus = _scanStatus.asStateFlow()
+    private val _status = MutableStateFlow<ScanStatus>(Stopped)
+    val status = _status.asStateFlow()
 
     private val _advertisements = MutableStateFlow<List<Advertisement>>(emptyList())
     val advertisements = _advertisements.asStateFlow()
 
-    fun startScan() {
-        if (_scanStatus.value == Started) return // Scan already in progress.
-        _scanStatus.value = Started
+    fun start() {
+        if (_status.value == Scanning) return // Scan already in progress.
+        _status.value = Scanning
```

Comment:
Unlikely to ever come up, but this can lead to race conditions.

```suggestion
        val priorStatus = _status.value
        if (priorStatus == Scanning || !priorStatus.compareAndSet(expected, Scanning)) return
```

Reply:
This function is only called from UI thread. So, it wouldn't suffer the race condition you're thinking, right?
I would definitely not want a race condition, but protecting against it is more code complexity for a sample app.
Perhaps marking as `@UiThread` would be a good middle ground?